bin/plugin/commands/common.js

```diff
@@ -133,3 +96,52 @@ exports.getModuleData = async ( modulesDir ) => {
 			return firstModule.slug.localeCompare( secondModule.slug );
 		} );
 };
+
+/**
+ * Returns the list of module data for all modules.
+ *
+ * @param {string} moduleFile Modules file.
+ *
+ * @return {[]WPModuleData} Module data list.
+ */
+exports.getModuleDataFromHeaders = ( moduleFile ) => {
+	// Populate slug and focus based on file path.
+	const moduleDir = path.dirname( moduleFile );
+	const moduleData = {
+		slug: path.basename( moduleDir ),
+		focus: path.basename( path.dirname( moduleDir ) ),
+	};
+
+	// Map of module header => object property.
+	const headers = {
+		'Module Name': 'name',
+		Description: 'description',
+		Experimental: 'experimental',
+	};
+
+	// Populate name, description and experimental based on module file headers.
+	const fileContent = fs.readFileSync( moduleFile, 'utf8' );
```

Comment:
This line won't be needed anymore in this function when you change it to receive the `moduleHeader` as a string.